### PR TITLE
Add murmur3_hash function

### LINKED ISSUE #4480: Add support for MurmurHash3 function

### PR DESCRIPTION
**Pull request by yashaswaj**
Fixes #4480

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/metadata/FunctionRegistry.java [hunk 1]
```diff
@@ -127,6 +127,7 @@
 import io.prestosql.operator.scalar.MapValues;
 import io.prestosql.operator.scalar.MathFunctions;
 import io.prestosql.operator.scalar.MultimapFromEntriesFunction;
+import io.prestosql.operator.scalar.MurmurHashFunction;
```
Comment (Praveen2112) [inline presto-main/src/main/java/io/prestosql/metadata/FunctionRegistry.java:130]: Can we use `Murmur3Hash32` which is available in Slice ?

**Review by Praveen2112 — COMMENTED** [review]
Thanks for working on this. Can we add doc for this function ?

File: presto-main/src/test/java/io/prestosql/operator/scalar/TestMurmurHashFunction.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import static io.prestosql.testing.SqlVarbinaryTestingUtil.sqlVarbinaryFromHex;
+
+public class TestMurmurHashFunction
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testMurmur3Hash()
+    {
+        assertFunction("murmur3_hash(1)", IntegerType.INTEGER, 1392991556);
```
Comment (electrum) [inline presto-main/src/test/java/io/prestosql/operator/scalar/TestMurmurHashFunction.java:35]: @martint should these return `varbinary` like the other hash functions?

File: presto-main/src/main/java/io/prestosql/operator/scalar/VarbinaryFunctions.java [hunk 3]
```diff
@@ -279,6 +280,14 @@ public static Slice sha512(@SqlType(StandardTypes.VARBINARY) Slice slice)
         return computeHash(Hashing.sha512(), slice);
     }
 
+    @Description("Compute murmur3 hash")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARBINARY)
+    public static Slice murmur3(@SqlType(StandardTypes.VARBINARY) Slice slice)
+    {
+        return Murmur3Hash128.hash(1, slice, 0, slice.length());
```
Comment (martint) [inline presto-main/src/main/java/io/prestosql/operator/scalar/VarbinaryFunctions.java:288]: Why seed = 1? Let's use the version with default seed:

```suggestion
        return Murmur3Hash128.hash(slice, 0, slice.length());
```

**Review by martint — CHANGES_REQUESTED** [review]
Just one minor comment. You'll need to adjust the unit tests, too.

File: presto-main/src/main/java/io/prestosql/operator/scalar/VarbinaryFunctions.java (hunk 3 above)
Comment (yashaswaj) [reply presto-main/src/main/java/io/prestosql/operator/scalar/VarbinaryFunctions.java:288]: With default seed, this method returns all zero bytes for empty string, thats why i added a seed.

Comment (martint) [reply presto-main/src/main/java/io/prestosql/operator/scalar/VarbinaryFunctions.java:288]: That's the expected hash for an empty string. For example, 

In python:
```
>>> import mmh3
>>> mmh3.hash('')
0
```

In Java:
```
System.out.println(Hashing.murmur3_32().hashBytes(new byte[0]).asInt());
-> 0
```

**Review by martint — APPROVED** [review]
Thanks! I'll merge it once the CI tests pass.

**Pull request merged**